src/components/table/sort.ts

```diff
@@ -1,13 +1,15 @@
-export function descendingComparator<T>(a: T, b: T, orderBy: keyof T, order: SortOrder): number {
-  // first attempt to parse into a numeric value and compare
-  const numCompare = descendingNumComparator(a, b, orderBy);
-  if (numCompare !== null) {
-    return numCompare;
-  }
+export function descendingComparator<T>(a: T, b: T, orderBy: keyof T | string, order: SortOrder): number {
+  const getValue = (obj: any, path: string) => {
+    return path.split('.').reduce((acc, part) => acc && acc[part], obj);
+  };
```

Comment:
Not really necessary, but here's how you could rewrite it if you want to be type safe and avoid using `any`. 

```suggestion
type UnknownObject = { [key: string]: unknown | UnknownObject };

export function descendingComparator<T extends UnknownObject>(
  a: T,
  b: T,
  orderBy: keyof T | string,
  order: SortOrder
): number {
  const getValue = (obj: unknown, path: string): unknown =>
    path
      .split('.')
      .reduce(
        (acc: unknown | UnknownObject, part: string): unknown | UnknownObject => acc && (acc as UnknownObject)[part],
        obj
      );
```